fhir-cql/src/main/java/com/ibm/fhir/cql/engine/model/FhirModelResolver.java

```diff
@@ -0,0 +1,516 @@
+            TemporalAccessor ta = (TemporalAccessor) result;
+            result = DateHelper.toCqlTemporal(ta);
+        } else if ( result instanceof byte[] ) {
+            result = Base64.getEncoder().encode((byte[])result);
+        } else if ( result instanceof Id ) {
+            result = ((Id)result).getValue();
+        }
+        return result;
+    }
+
+    @Override
+    public Object getContextPath(String contextType, String targetType) {
+        Object result = null;
+        if (targetType != null && contextType != null) {
+
+            if (contextType != null && !(contextType.equals("Unspecified") || contextType.equals("Population"))) {
+                if (targetType != null && contextType.equals(targetType)) {
+                    result = "id";
+                } else {
+                    Class<?> clazz = resolveType(targetType);
+
+                    if (clazz != null) {
+                        Set<String> possibleAnswers = new HashSet<String>();
+                        getReferenceTargets(clazz, contextType, "", possibleAnswers);
+
+                        if (possibleAnswers.isEmpty()) {
+                            throw new IllegalArgumentException(String.format("Unable to determine context path value for context '%s' in type '%s'", contextType, targetType));
+                        } else if (possibleAnswers.size() == 1) {
+                            result = possibleAnswers.iterator().next();
+                        } else {
+                            if (contextType.equals("Patient")) {
+                                String[] bestGuesses = new String[] { "subject", "patient", "beneficiary" };
+                                for (String guess : bestGuesses) {
+                                    if (possibleAnswers.contains(guess)) {
+                                        result = guess;
+                                        break;
+                                    }
+                                }
+                            }
+                            if (result == null) {
+                                throw new IllegalArgumentException(String.format("ContextPath '%s' is ambiguous for type '%s'. Possible answers are '%s'.", contextType, targetType, possibleAnswers.toString()));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    private void getReferenceTargets(Class<?> clazz, String contextType, String prefix, Set<String> possibleAnswers) {
+        while (clazz != null && clazz.getPackage().getName().startsWith(this.packageName)) {
+
+            getReferenceTargetFromFields(clazz, contextType, prefix, possibleAnswers);
+
+            for (Class<?> subclass : clazz.getDeclaredClasses()) {
+                if (!subclass.getName().endsWith("$Builder")) {
+                    String newPrefix = getPrefixed(prefix, StringUtils.uncapitalize(subclass.getSimpleName()));

```

Comment:
```suggestion
                    String newPrefix = getPrefixed(prefix, StringUtils.uncapitalize(subclass.getSimpleName()));

```
Any reason not to just use toLowerCase?